frontend/dockerfile/dockerfile_secrets_test.go

```diff
@@ -80,3 +82,63 @@ RUN --mount=type=secret,required,id=mysecret foo
 	require.Error(t, err)
 	require.Contains(t, err.Error(), "secret mysecret: not found")
 }
+
+func testSecretAsEnviron(t *testing.T, sb integration.Sandbox) {
+	integration.SkipOnPlatform(t, "windows")
+	f := getFrontend(t, sb)
+
+	dockerfile := []byte(`
+FROM busybox
+RUN --mount=type=secret,id=mysecret,env [ "$mysecret" == "pw" ] && [ -f /run/secrets/mysecret ] || false
```

Comment:
I'm not sure if this is needed in Dockerfile syntax. Would be more clearer if `env` always has a string value.